onnxruntime/core/providers/cpu/containers.h

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#if defined(USE_MIMALLOC_STL_ALLOCATOR)
+#include <mimalloc.h>
+#endif
+
+namespace onnxruntime {
+
+template <class T>
+struct BFCArenaAllocator {
```

Comment:
This is not necessarily a BFC arena allocator, right? I mean it could wrap any allocator to conform to the STL allocator concept. The name is a bit confusing. We already have a BFCArena class that derives from IArenaAllocator and now we've this class. Moreover, this name doesn't tell me that it's intended to be used as a STL allocator. Should we rename this to OrtStlAllocator or CustomStlAllocator?